packages/EasyWebhook/docs/middleware.md

```diff
@@ -0,0 +1,217 @@
+and if so, resets its `$currentAttempt` and its `$status` to `pending`, which allows the webhook to be rerun.
+
+### `ResetStoreMiddleware`
+
+This middleware resets the webhook store and webhook results store. See [Stores](stores.md) for more information.
+
+Note that the stores must implement `EonX\EasyWebhook\Interfaces\Stores\ResetStoreInterface`. Of the stores provided by
+the EasyWebhook package, only the array stores support reset.
+
+### `SendAfterMiddleware`
+
+This middleware checks the `$sendAfter` property of the webhook, and if it exists and is after the current date and
+time, it simply stores the webhook in the configured store, and does not proceed with any further middleware processing.
+You can initiate sending of delayed webhooks via the console command `easy-webhooks:send-due-webhooks`. See
+[Console](console.md) for more information.
+
+Note that the webhook store must implement `EonX\EasyWebhook\Interfaces\Stores\SendAfterStoreInterface` in order to be
+able to find due webhooks. Of the stores provided by the EasyWebhook package, only the Doctrine DBAL webhook store
+supports finding due webhooks.
+
+### `SendWebhookMiddleware`
+
+This middleware sends the webhook HTTP request and returns the result in a [WebhookResult](webhook-result.md) object.
+
+### `SignatureHeaderMiddleware`
+
+This middleware sets the `X-Webhook-Signature` header of the webhook HTTP request. The header contains a SHA-256 HMAC
+(Hash-based Message Authentication Code) constructed by hashing the request body with either the webhook's `$secret`
+property (if it exists) or the default secret.
+
+*This is configure once middleware.*
+
+### `StatusAndAttemptMiddleware`
+
+This middleware updates the webhook status and current attempt after a response has been received from the webhook HTTP
+request.
+
+The current attempt is incremented.
+
+The status can be set to one of:
+
+- `success`: The webhook was successfully sent.
+- `failed`: The webhook sending failed and the webhook has reached the maximum number of attempts allowed
+- `failed_pending_retry`: The webhook sending failed but the webhook has not reached the maximum number of attempts
+  allowed
+
+### `StoreMiddleware`
+
+This middleware stores the webhook and webhook result in the configured stores after a response has been received from
+the webhook HTTP request. See [Stores](stores.md) for more information.
+
+## Middleware stack
+
+The following table show the middleware stack in priority order, with summaries of their actions:
+
+| Middleware | Action forward :arrow_down: | Action back :arrow_up: |
+| ---------- | --------------------------- | ---------------------- |
+| `LockMiddleware` | Lock webhook | Unlock webhook |
+| `RerunMiddleware` | If rerun allowed, reset status and current attempt<br/>If not allowed, throw exception | |
+| *Begin core middleware*<br/>*(processed in priority order)* | | |
```

Comment:
Like I said in a comment above, "core middleware" are the one from the package that CANNOT be removed to guarantee the package to work as expected.

So here, the core middleware don't begin but end 😛

Reply:
Here the "Before Core Middleware" end.